src/network/networkctl.c

```diff
@@ -242,6 +248,43 @@ static const LinkInfo* link_info_array_free(LinkInfo *array) {
 }
 DEFINE_TRIVIAL_CLEANUP_FUNC(LinkInfo*, link_info_array_free);
 
+static int acquire_azure_cloud_metadata(NetworkCloudManager **ret) {
+        _cleanup_(network_cloud_manager_freep) NetworkCloudManager *m = NULL;
+        _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+        int r;
+
+        if (!streq("microsoft", virtualization_to_string(detect_virtualization())))
+                return -ENOTSUP;
+
+        r = network_cloud_manager_new(CLOUD_PROVIDER_AZURE, "http://169.254.169.254/metadata/instance/network?api-version=2017-04-02", &m);
```

Comment:
Maybe define this URL in a `#define` somewhere and add a comment explaining where it comes from.